### ISSUE TITLE
Support ActivityPub

### ISSUE THREAD

**Issue by evanp** [idno/Known#1701]
Known should support the [ActivityPub](https://www.w3.org/TR/activitypub/) ("AP") standard. It is the fruit of the last several years of work at the W3C. It provides:

- a client-to-server API for publishing content, reacting to content, and maintaining social networks
- a server-to-server API for distributing activities between servers

Projects like pump.io and GNU MediaGoblin are implementing the standard. More info at [ActivityPub network](https://www.w3.org/wiki/Socialwg/ActivityPub_network).

**Comment by mapkyca** [comment]
Cool, but I certainly don't have time to get to this atm... PRs welcomed of course!

**Comment by akuckartz** [comment]
:+1:

**Comment by benwerd** [comment]
I'm definitely in support of this.

**Comment by benwerd** [comment]
Confirming that I'm taking this on.

**Comment by physcocode** [comment]
@benwerd I am in 👍

**Comment by benwerd** [comment]
Bumping this to reconfirm. All help appreciated!

**Comment by jgmac1106** [comment]
Sounds like an opportunity for someone to work on at IWC Nurnberg and Berlin. Lot of activity pub plumbing created by Mathias and Jacky that we might be to used as base for Idno plugin. Though WordPress is still OStatus .

**Comment by physcocode** [comment]
Been following up indieweb moment and known

**Comment by jgmac1106** [comment]
Can we just use bridgy-fed? Lot of cool development already done there: https://github.com/snarfed/bridgy-fed

**Comment by benwerd** [comment]
Yes and no. Yes for individual installs - but I don't want anyone who's using Known for an enterprise-level multi-user site to depend on Bridgy (mostly because I think that's not fair on Bridgy).

**Comment by almereyda** [comment]
As an active Idno/Known user since early days, I might be interested to join folks in Berlin during https://indieweb.org/2018/Berlin

Is someone planning to attend?

**Comment by mapkyca** [comment]
<a href="https://github.com/idno/Known/issues/1701#issuecomment-431086589" class="u-in-reply-to"></a>Wasn't planning to, but I do love Berlin, so I might have a look at flights this weekend.

Would certainly be interested in seeing some PRs and contributions.

**Comment by jgmac1106** [comment]
There is a large contingency going that will be working on Activity Pub in Berlin...not so. I will be attending remotely, tried to find the money,  but will also be pretty useless in terms of AP..

That being said I am happy to help with Activity Pub in anyway I can for Nurenberg or Berlin. Will also be trying to create an idno webmention badgs plugin.

Related repos:
https://github.com/pfefferle/wordpress-activitypub
https://github.com/aaronpk/Nautilus

Maybe we just convince @snarfed to release an paid enterprise version of bridgy.

**Comment by strypey** [comment]
AFAIK bridgy is all free code. Couldn't you set up your own instance(s) for your enterprise customers?

**Comment by herbsmn** [comment]
I'd love to see this feature implimented as well.

**Comment by poVoq** [comment]
https://github.com/pterotype-project/activitypub-php

Maybe a starting point? At least with WordPress it works reasonably well.

**Comment by benwerd** [comment]
This is now a funded issue here: https://github.com/idno/known/issues/2615

**Comment by lindner** [comment]
Exciting!  Glad to see this experiment in action.

**Comment by Cambridgeport90** [comment]
I can’t wait to see this… after all, I’m thinking of returning to Known once things settle down on my end, so will be tracking progress. I can’t code in PHP worth a dang and a half, but I can test.

**Comment by ipranjal** [comment]
@benwerd @mapkyca just a question , should ActivityPub implemented in core or needs to be implemented as separate plugin ?

**Comment by benwerd** [comment]
I believe it should be implemented in core.

Ben Werdmuller
( http://goog_1933028737 )
werd.io ( http://werd.io ) | @benwerd ( https://twitter.com/benwerd )

(he/him)

**Comment by Cambridgeport90** [comment]
Definitely.

**Comment by ipranjal** [comment]
little bit of dive made it obvious that it needs to be implemented in the core , here is the rough draft https://github.com/idno/known/issues/2615#issuecomment-991778488

**Comment by ipranjal** [comment]
should close this discussion in favour of #2615

**Comment by almereyda** [comment]
Agreed, this can bee closed, now that #2615 has more traction.

*Edit:* Despite issue number 1701 is lower (=older) and has a nice Star Trek flair to it …
